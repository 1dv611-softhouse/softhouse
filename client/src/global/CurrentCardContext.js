import React, { useState, createContext } from "react";

export const CurrentCardContext = createContext();

export const CurrentCardProvider = (props) => {
  const [currentCard, setCurrentCard] = useState({});
  return (
    <CurrentCardContext.Provider value={{ currentCard, setCurrentCard }}>
      {props.children}
    </CurrentCardContext.Provider>
  );
};
