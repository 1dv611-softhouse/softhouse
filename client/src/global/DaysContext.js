import React, { useState, createContext } from 'react'

export const DaysContext = createContext()

export const DaysProvider = (props) => {
  const [days, setDays] = useState([])
  return (
    <DaysContext.Provider value={{ days, setDays }}>
      {props.children}
    </DaysContext.Provider>
  )
}
