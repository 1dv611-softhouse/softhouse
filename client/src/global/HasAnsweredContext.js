import React, { useState, createContext } from 'react'

export const HasAnsweredContext = createContext()

export const HasAnsweredProvider = (props) => {
  const [hasAnswered, setHasAnswered] = useState(true)
  return (
    <HasAnsweredContext.Provider value={{ hasAnswered, setHasAnswered }}>
      {props.children}
    </HasAnsweredContext.Provider>
  )
}
