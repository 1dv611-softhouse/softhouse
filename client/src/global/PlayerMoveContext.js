import React, { useState, createContext } from 'react'

export const PlayerMoveContext = createContext()

export const PlayerMoveProvider = (props) => {
  const [currentPlayerMove, setPlayerMove] = useState({})
  return (
    <PlayerMoveContext.Provider value={{ currentPlayerMove, setPlayerMove }}>
      {props.children}
    </PlayerMoveContext.Provider>
  )
}
