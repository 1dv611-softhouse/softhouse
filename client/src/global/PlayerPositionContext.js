import React, { useState, createContext } from "react";

export const PlayerPositionContext = createContext();

export const PlayerPositionProvider = (props) => {
  const [currentPositionValue, setCurrentPositionValue] = useState(1);
  if (currentPositionValue > 22) {
    const newValue = currentPositionValue - 22;
    setCurrentPositionValue(newValue);
  }
  return (
    <PlayerPositionContext.Provider
      value={{ currentPositionValue, setCurrentPositionValue }}
    >
      {props.children}
    </PlayerPositionContext.Provider>
  );
};
