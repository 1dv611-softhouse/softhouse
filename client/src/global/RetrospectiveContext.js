import React, { useState, createContext } from 'react'

export const RetrospectiveContext = createContext()

export const RetrospectiveProvider = (props) => {
  const [retrospective, setRetrospective] = useState(false)
  return (
    <RetrospectiveContext.Provider value={{ retrospective, setRetrospective }}>
      {props.children}
    </RetrospectiveContext.Provider>
  )
}
