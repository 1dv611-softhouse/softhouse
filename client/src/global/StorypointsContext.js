import React, { useState, createContext } from 'react'

export const StorypointsContext = createContext()

export const StorypointsProvider = (props) => {
  const [currentStorypoints, setCurrentStorypoints] = useState(200)
  return (
    <StorypointsContext.Provider
      value={{ currentStorypoints, setCurrentStorypoints }}
    >
      {props.children}
    </StorypointsContext.Provider>
  )
}
