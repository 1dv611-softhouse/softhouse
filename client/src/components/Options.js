import "../styles/options.css";
import { useContext } from "react";
import socials from "../pictures/socials/images2";
import { HeaderContext } from "../global/HeaderContext";
import { HamburgerContext } from "../global/HamburgerContext";

function Options() {
  const { currentComponent, setCurrentComponent } = useContext(HeaderContext);
  const { active, setActive } = useContext(HamburgerContext);

  const socialMediaLinks = [
    "https://www.facebook.com/softhouseconsulting",
    "",
    "https://www.instagram.com/softhouseconsulting/",
    "",
    "",
  ];

  const closeMenu = (value) => {
    setCurrentComponent(value);
    setActive(!active);
  };

  return (
    <div className="options-container">
      <ul>
        <li>
          <p data-testid="profile-option" onClick={() => closeMenu("profile")}>Profile</p>
        </li>
        <li>
          <p onClick={() => closeMenu("settings")}>Settings</p>
        </li>
        <li>
          <p onClick={() => closeMenu("highscore")}>HighScore</p>
        </li>
        <li>
          <p onClick={() => closeMenu("rules")}>Rules</p>
        </li>
        {/* <li>
          <p onClick={() => closeMenu("logout")}>Logout</p>
        </li> */}
      </ul>

      <div className="social-media-icons">
        {socials.map((img, index) => {
          return (
            <a href={socialMediaLinks[index]}>
              <img src={img} alt="Social media"/>
            </a>
          );
        })}
      </div>
    </div>
  );
}

export default Options;
