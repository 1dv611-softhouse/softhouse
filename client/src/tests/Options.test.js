import Options from '../components/Options'
import ReactDOM from 'react-dom'
import { render, fireEvent } from '@testing-library/react'
import { HeaderProvider } from '../global/HeaderContext'
import { HamburgerProvider } from '../global/HamburgerContext'

import Enzyme  from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { shallow } from 'enzyme'
import { mount } from 'enzyme'

Enzyme.configure({ adapter: new Adapter() });

const wrapperShallow = shallow(
    <HeaderProvider>
        <HamburgerProvider>
            <Options />
        </HamburgerProvider>
    </HeaderProvider>
    )   
    
    const wrapperMount = mount(
        <HeaderProvider>
        <HamburgerProvider>
            <Options />
        </HamburgerProvider>
    </HeaderProvider>
    )

describe('Testing the options component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

      it('should render without crashing', () => {
        const div = document.createElement('div')

        ReactDOM.render(
            <HeaderProvider>
                <HamburgerProvider>
                    <Options />
                </HamburgerProvider>
            </HeaderProvider>, div
        )
    })

    it('should have 1 options-container', () => {
        expect(wrapperMount.find('.options-container').length).toEqual(1)
    })

    it('There should be 5 options to choose from', () => {
        expect(wrapperMount.find('p').length).toEqual(5)
     })

     it('Should contain a P element with textcontent: Profile', () => {
        const { getByTestId } = render(
            <HeaderProvider>
                <HamburgerProvider>
                    <Options />
                </HamburgerProvider>
            </HeaderProvider>
        )

        const pTag = getByTestId('profile-option')
        expect(pTag.textContent).toBe('Profile')
     })
})

