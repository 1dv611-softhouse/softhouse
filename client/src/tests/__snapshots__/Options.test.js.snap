// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the options component should match the snapshot 1`] = `"<div class=\\"options-container\\"><ul><li><p data-testid=\\"profile-option\\">Profile</p></li><li><p>Settings</p></li><li><p>HighScore</p></li><li><p>Rules</p></li><li><p>Logout</p></li></ul><div class=\\"social-media-icons\\"><a href=\\"https://www.facebook.com/softhouseconsulting\\"><img src=\\"1.png\\" alt=\\"Social media\\"/></a><a href=\\"\\"><img src=\\"2.png\\" alt=\\"Social media\\"/></a><a href=\\"https://www.instagram.com/softhouseconsulting/\\"><img src=\\"3.png\\" alt=\\"Social media\\"/></a><a href=\\"\\"><img src=\\"4.png\\" alt=\\"Social media\\"/></a><a href=\\"\\"><img src=\\"5.png\\" alt=\\"Social media\\"/></a></div></div>"`;
